### PR TITLE
Fix GGUF-K RTN routing and fake eval dtype normalization

### PR DESCRIPTION
**Pull request by n1ck-guo**
## Description

Please briefly describe your main changes, the motivation.

## Type of Change

Bug fix

## Related Issues

Fixes or relates to #

## Checklist Before Submitting

- [ ] My code has been tested locally.
- [ ] Documentation has been updated as needed.
- [ ] New or updated tests are included where applicable.
- [ ] The CUDA CI has passed. You can trigger it by commenting `/azp run Unit-Test-CUDA-AutoRound`.

### CODE REVIEW

File: auto_round/eval/evaluation.py [hunk 1]
```diff
@@ -226,6 +259,8 @@ def prepare_model_for_eval(model, device_map, eval_model_dtype):
 
     from auto_round.utils import detect_device
 
+    model = _normalize_model_eval_dtype(model, eval_model_dtype)
+
```
[the remark this answers is not in the document]
Comment (n1ck-guo) [reply auto_round/eval/evaluation.py]: @copilot apply changes based on [this feedback](https://github.com/intel/auto-round/pull/1807#discussion_r3231032257)

File: auto_round/eval/evaluation.py [hunk 2]
```diff
@@ -21,6 +21,39 @@
...
+            floating_dtypes.add(buffer.dtype)
+    return floating_dtypes
+
+
+def _normalize_model_eval_dtype(model, eval_model_dtype):
+    import torch
+
+    floating_dtypes = _collect_model_floating_dtypes(model)
+    if not floating_dtypes:
+        return model
```
Comment (wenhuach21) [inline auto_round/eval/evaluation.py:40]: There is no need to parse all the parameters. For any floating-point dtype, simply return and route to the corresponding logic.

**Review by wenhuach21 — APPROVED** [review]
